lib/xo.ts

```diff
@@ -251,6 +251,30 @@ export class Xo {
 		}
 	}
 
+	/**
+	 * Checks if any directly specified files are ignored and throws an error.
+	 * @private
+	 */
+	async checkForIgnoredDirectFiles(globs: string[]): Promise<void> {
+		const {flatOptions} = await resolveXoConfig(this.linterOptions);
+		const combinedIgnores = [
+			...defaultIgnores,
+			...arrify(this.baseXoConfig.ignores),
+			...flatOptions.flatMap(config => config.ignores ? arrify(config.ignores) : []),
```

Comment:
I don't think this is quite correct logic. If a flatConfig can has ignores AND files, then we only want to apply those ignores to the files in that config...